cobalt/web/agent.cc

```diff
@@ -535,6 +544,12 @@ void Agent::Stop() {
     context()->service_worker_context()->UnregisterWebContext(context());
   }
 
+  watchdog::Watchdog* watchdog = watchdog::Watchdog::GetInstance();
+  if (watchdog && watchdog_registered_) {
```

Comment:
I would leave it as just `if (watchdog) {` rather than have this extra safety check. We should not unregister before registering. The unregister will handle it if we erroneously unregister before registering and it gives us an appropriate error message.